src/libopenrave-core/jsonparser/jsonreader.cpp

```diff
@@ -469,6 +477,8 @@ class JSONReader
             if( !pRobot ) {
                 return false;
             }
+            // for extracting a single body from file, need to assign a random body id to not conflict with what's already in environment
+            pRobotBaseInfo->_id = str(boost::format("body%d")%utils::GetNanoTime());
```

Comment:
@ziyan same as above

Reply:
For now, I think empty id is okay for this case. Resolved.